src/main/java/org/graylog/aws/inputs/cloudtrail/notifications/CloudtrailSNSNotificationParser.java

```diff
@@ -1,42 +1,42 @@
 package org.graylog.aws.inputs.cloudtrail.notifications;
 
 import com.amazonaws.services.sqs.model.Message;
+import com.fasterxml.jackson.databind.DeserializationFeature;
 import com.fasterxml.jackson.databind.ObjectMapper;
-import com.google.common.collect.Lists;
 import org.graylog.aws.inputs.cloudtrail.json.CloudtrailWriteNotification;
 import org.graylog.aws.json.SQSMessage;
 
 import java.io.IOException;
+import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 
 public class CloudtrailSNSNotificationParser {
-    private final ObjectMapper om;
-
-    public CloudtrailSNSNotificationParser() {
-        om = new ObjectMapper();
-    }
+    private final ObjectMapper objectMapper = new ObjectMapper().disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
 
     public List<CloudtrailSNSNotification> parse(Message message) {
-        List<CloudtrailSNSNotification> notifications = Lists.newArrayList();
 
         try {
-            SQSMessage envelope = om.readValue(message.getBody(), SQSMessage.class);
+            final SQSMessage envelope = objectMapper.readValue(message.getBody(), SQSMessage.class);
 
             if (envelope.message == null) {
                 return Collections.emptyList();
             }
 
-            CloudtrailWriteNotification notification = om.readValue(envelope.message, CloudtrailWriteNotification.class);
+            final CloudtrailWriteNotification notification = objectMapper.readValue(envelope.message, CloudtrailWriteNotification.class);
```

Comment:
Does this need to be done in `TreeReader.java` as well, it is using a custom ObjectMapper instance as well.

I'm wondering why the `@JsonIgnoreProperties(ignoreUnknown = true)` that the DTOs have on them don't work as expected?

Reply:
The parsing failed with `CloudtrailWriteNotification` (which doesn't have this annotation), not with `SQSMessage` (which does).